internal/activities/add_premis_objects.go

```diff
@@ -0,0 +1,68 @@
+package activities
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+
+	"github.com/google/uuid"
+
+	"github.com/artefactual-sdps/preprocessing-sfa/internal/premis"
+)
+
+const AddPREMISObjectsName = "add-premis-objects"
+
+type AddPREMISObjectsParams struct {
+	PREMISFilePath string
+	ContentPath    string
+}
+
+type AddPREMISObjectsResult struct{}
+
+type AddPREMISObjectsActivity struct{}
+
+func NewAddPREMISObjects() *AddPREMISObjectsActivity {
+	return &AddPREMISObjectsActivity{}
+}
+
+func (md *AddPREMISObjectsActivity) Execute(
+	ctx context.Context,
+	params *AddPREMISObjectsParams,
+) (*AddPREMISObjectsResult, error) {
+	// Get subpaths of files in transfer.
+	subpaths, err := premis.FilesWithinDirectory(params.ContentPath)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create parent directory, if necessary.
+	mdPath := filepath.Dir(params.PREMISFilePath)
+	if err := os.MkdirAll(mdPath, 0o700); err != nil {
+		return nil, err
+	}
+
+	for _, subpath := range subpaths {
+		doc, err := premis.ParseOrInitialize(params.PREMISFilePath)
```

Comment:
Do we really need to write a separate premis.xml for each subpath here?  I think the final result is a single premis.xml in the PIP, isn't it?

Reply:
Fixed!